Engine/ResourceManagement/Manager/StateMachineManager.cpp

```diff
@@ -4,44 +4,50 @@
 #include "Module/ModuleFileSystem.h"
 #include "Module/ModuleResourceManager.h"
 #include "ResourceManagement/Resources/StateMachine.h"
-std::shared_ptr<StateMachine> StateMachineManager::Load(const std::string & file_path)
+
+FileData StateMachineManager::Binarize(StateMachine* state_machine)
 {
-	if (!App->filesystem->Exists(file_path.c_str()))
-	{
-		APP_LOG_ERROR("Error loading State Machine %s.", file_path.c_str());
-		return nullptr;
-	}
+	Config state_machine_config;
+	state_machine->Save(state_machine_config);
+
+	std::string serialized_state_machine_string;
+	state_machine_config.GetSerializedString(serialized_state_machine_string);
 
-	APP_LOG_INFO("Loading State Machine  %s.", file_path.c_str());
-	size_t size;
-	char * data = App->filesystem->Load(file_path.c_str(), size);
+	char* state_machine_bytes = new char[serialized_state_machine_string.size() + 1];
+	memcpy(state_machine_bytes, serialized_state_machine_string.c_str(), serialized_state_machine_string.size() + 1);
+
+	FileData state_machine_data{ state_machine_bytes, serialized_state_machine_string.size() + 1 };
+	return state_machine_data;
+}
+
+std::shared_ptr<StateMachine> StateMachineManager::Load(uint32_t uuid, const FileData& resource_data)
```

Comment:
free(data) missing at the end

Reply:
I will refactor that comment. There was a free(data) before in order to avoid memory leaks, now how you avoid that?

Reply:
This method is called from `ModuelResourceManagement`, in line 114 of header file, in function `Load`. After that, the data is freed in line 116. 
```
loaded_resource = ResourceManagement::Load<T>(uuid, exported_file_data);
((char*)exported_file_data.buffer);
```
Related to your previous comment, all the data management responsability is held by class `ModuelResourceManagement` in order to not have to remember to `free(data)`in every Importer/Manager.